### PR TITLE
fix(policies): raise clear error on checkpoint/model dimension mismatch

### LINKED ISSUE #2963: pi05_base pretrained model has action/state dimension mismatch with lerobot/libero dataset, causing official Quick Start example to fail
### Ticket Type

🐛 Bug Report (Something isn't working)

### Environment & System Info

```Shell
lerobot-info
- LeRobot version: 0.4.4
- Platform: Linux-5.15.0-124-generic-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface Hub version: 0.35.3
- Datasets version: 4.1.1
- Numpy version: 2.2.6
- FFmpeg version: 4.4.2-0ubuntu0.22.04.1
- PyTorch version: 2.7.1+cu126
- Is PyTorch built with CUDA support?: True
- Cuda version: 12.6
- GPU model: NVIDIA GeForce RTX 3090
- Using GPU in script?: <fill in>
- lerobot scripts: ['lerobot-calibrate', 'lerobot-dataset-viz', 'lerobot-edit-dataset', 'lerobot-eval', 'lerobot-find-cameras', 'lerobot-find-joint-limits', 'lerobot-find-port', 'lerobot-imgtransform-viz', 'lerobot-info', 'lerobot-record', 'lerobot-replay', 'lerobot-setup-can', 'lerobot-setup-motors', 'lerobot-teleoperate', 'lerobot-train', 'lerobot-train-tokenizer']
```

### Description

According to the Quick Start code in the official [lerobot/pi05_base README](https://huggingface.co/lerobot/pi05_base#quick-start-inference-on-a-real-batch), users are instructed to run inference with:

python
```
model_id = "lerobot/pi05_base"
policy = PI05Policy.from_pretrained(model_id)
dataset = LeRobotDataset("lerobot/libero")
```
However, this fails immediately with state/action dimension mismatch errors. The pretrained pi05_base model expects 32‑dimensional states and actions (e.g., as in Aloha robots), while lerobot/libero actually provides 8‑dimensional states and actions.

### Context & Reproduction

1. Install the latest lerobot: pip install lerobot
2. Run the exact script from the README (or the minimal version below):
```

python
import torch
from lerobot.datasets.lerobot_dataset import LeRobotDataset
from lerobot.policies.pi05 import PI05Policy

model_id = "lerobot/pi05_base"
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
policy = PI05Policy.from_pretrained(model_id).to(device).eval()

dataset = LeRobotDataset("lerobot/libero")
episode_index = 0
from_idx = dataset.meta.episodes["dataset_from_index"][episode_index]
frame_index = from_idx
frame = dict(dataset[frame_index])

# Preprocess and select action will also fail, but the dimension error appears during from_pretrained
# if the config is not manually fixed.
```
3. Expected behavior: Model loads successfully and inference runs.
Actual behavior: The following errors appear (full traceback attached):
```

size mismatch for model.action_in_proj.weight: copying a param with shape torch.Size([1024, 32]) from checkpoint, the shape in current model is torch.Size([1024, 8]).
size mismatch for model.action_out_proj.weight: copying a param with shape torch.Size([32, 1024]) from checkpoint, the shape in current model is torch.Size([8, 1024]).
size mismatch for model.action_out_proj.bias: copying a param with shape torch.Size([32]) from checkpoint, the shape in current model is torch.Size([8]).


```

### Relevant logs or stack trace

```Shell
root@instance-8e4pi2j3:/lerobot# python infer_local_model.py 
The PI05 model is a direct port of the OpenPI implementation. 
This implementation follows the original OpenPI structure for compatibility. 
Original implementation: https://github.com/Physical-Intelligence/openpi
Loading model from: /workspace/pi05_base
✓ Loaded state dict from model.safetensors
WARNING:root:Vision embedding key might need handling: paligemma_with_expert.paligemma.model.vision_tower.vision_model.embeddings.patch_embedding.bias
WARNING:root:Vision embedding key might need handling: paligemma_with_expert.paligemma.model.vision_tower.vision_model.embeddings.patch_embedding.weight
Remapped: action_in_proj.bias -> model.action_in_proj.bias
Remapped: action_in_proj.weight -> model.action_in_proj.weight
Remapped: action_out_proj.bias -> model.action_out_proj.bias
Remapped: action_out_proj.weight -> model.action_out_proj.weight
Remapped: paligemma_with_expert.gemma_expert.lm_head.weight -> model.paligemma_with_expert.gemma_expert.lm_head.weight
Remapped: paligemma_with_expert.gemma_expert.model.layers.0.input_layernorm.dense.bias -> model.paligemma_with_expert.gemma_expert.model.layers.0.input_layernorm.dense.bias
Remapped: paligemma_with_expert.gemma_expert.model.layers.0.input_layernorm.dense.weight -> model.paligemma_with_expert.gemma_expert.model.layers.0.input_layernorm.dense.weight
Remapped: paligemma_with_expert.gemma_expert.model.layers.0.mlp.down_proj.weight -> model.paligemma_with_expert.gemma_expert.model.layers.0.mlp.down_proj.weight
Remapped: paligemma_with_expert.gemma_expert.model.layers.0.mlp.gate_proj.weight -> model.paligemma_with_expert.gemma_expert.model.layers.0.mlp.gate_proj.weight
Remapped: paligemma_with_expert.gemma_expert.model.layers.0.mlp.up_proj.weight -> model.paligemma_with_expert.gemma_expert.model.layers.0.mlp.up_proj.weight
Remapped 812 state dict keys
Warning: Could not remap state dict keys: Error(s) in loading state_dict for PI05Policy:
        Missing key(s) in state_dict: "model.paligemma_with_expert.paligemma.model.language_model.embed_tokens.weight". 
        size mismatch for model.action_in_proj.weight: copying a param with shape torch.Size([1024, 32]) from checkpoint, the shape in current model is torch.Size([1024, 8]).
        size mismatch for model.action_out_proj.weight: copying a param with shape torch.Size([32, 1024]) from checkpoint, the shape in current model is torch.Size([8, 1024]).
        size mismatch for model.action_out_proj.bias: copying a param with shape torch.Size([32]) from checkpoint, the shape in current model is torch.Size([8]).
```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

### Additional Info / Workarounds

_No response_

### PR DESCRIPTION
**Pull request by takashi-jp**
## Summary

Loading a pretrained checkpoint into a policy whose action/state feature dimensions differ from the checkpoint's currently surfaces a raw PyTorch error like:

```
size mismatch for model.action_in_proj.weight: copying a param with shape torch.Size([1024, 32]) from checkpoint, the shape in current model is torch.Size([1024, 8])
```

with no indication of what caused it or how to fix it. This is exactly what happens in #2963: `pi05_base` was trained on Aloha's 32-dim action/state, and the official Quick Start example pairs it with the 8-dim `lerobot/libero` dataset.

This PR catches that `RuntimeError` in `PreTrainedPolicy._load_as_safetensor` (`src/lerobot/policies/pretrained.py`) and re-raises it with an explanation of the likely cause and a suggested fix. The fix is generic — it applies to any policy/checkpoint combination with mismatched dimensions, not just `pi05_base` + `lerobot/libero`.

Closes #2963

## Test plan

- [x] Added `test_load_pretrained_dimension_mismatch_error` in `tests/policies/test_policies.py`, which saves an `act` checkpoint and loads it into a config with doubled action dims, asserting the new error message is raised.
- [x] `uv run pytest tests/policies/test_policies.py -svv --maxfail=10` — 16 passed, 11 skipped (pre-existing skips, unrelated to this change).
- [x] `uv run ruff check` / `ruff format --check` on changed files — clean.

### CODE REVIEW

**Review by A-lost-unicorn — COMMENTED** [review]
I reviewed the checkpoint-loading path and the regression test against #2963. This is a well-scoped improvement:

- The catch is limited to `RuntimeError`s containing `size mismatch`; unrelated runtime errors are re-raised unchanged.
- Exception chaining preserves the original safetensors/PyTorch details while adding actionable compatibility guidance.
- The regression test exercises the actual save/load path with a deliberately mismatched action dimension, instead of only asserting a helper or message in isolation.

A non-blocking tradeoff is that the string match depends on the current loader error wording. Given the lack of a structured mismatch type on this path and the safe re-raise fallback, this seems reasonable.

I don't see any blocking issues from my side. Thanks for the clear, focused fix.
